File: leetcode/go/2244.minimum-rounds-to-complete-all-tasks.go
func minimumRounds(tasks []int) int {
    counts := make(map[int]int, len(tasks))
    for _, t := range tasks {
        if _, ok := counts[t]; !ok {
            counts[t] = 0
        }
        counts[t]++
    }
    rounds := 0
    for _,c := range counts {
        if c == 1 {
            return -1
        }
        rounds += (c + 2)/3
    }
    return rounds
}

File: leetcode/rust-leetcode/src/2244.minimum-rounds-to-complete-all-tasks.rs
impl Solution {
    pub fn minimum_rounds(tasks: Vec<i32>) -> i32 {
        use std::collections::HashMap;
        let mut counts = HashMap::new();
        for t in tasks {
            *counts.entry(t).or_insert(0) += 1;
        }
        let mut rounds = 0;
        for (t, c) in counts.into_iter() {
            if c == 1 {
                return -1;
            }
            rounds += (c + 2)/3;
        }
        rounds
    }
}

